### ISSUE TITLE
Auto-delete users after time of inactivity

### ISSUE THREAD

**Issue by nilsnolde** [engelsystem/engelsystem#981]
We use the engelsystem for an annual conference (FOSSGIS) and many users/angels don't change over time, so we'd like to keep active accounts but also clean inactive accounts after some time, so it doesn't get too bloated.

Off the top of your head is it fine to remove from the `users` table? Is it linked to other tables by smth like `ON DELETE CASCADE`? If you don't know off the top of your head, I'll try locally at some point.

**Comment by MyIgel** [comment]
Hi @nilsnolde, al relations should delete on cascade so deleting users from the users table based on their last login time should be sufficient here as thats what the normal delting users page does. Deleting users after an event that where not active since the last event should be possible as we have the last action date.
If you find any issues with that feel free to create a bug report :)

**Comment by nilsnolde** [comment]
Ok awesome, thanks a bunch!

**Comment by msquare** [comment]
Hi, let me add some comments to this:
As @MyIgel mentioned, the cascades should delete all user related data, including their shifts, worklogs, etc.
When deleting by database, you get no log entry and also the user does not receive the "your account has been deleted" e-mail confirmation.
In fact we received like 10 mailserver errors from these e-mails from your instance and got blocked by a few mail providers because you didn't configure a correct sender e-mail-address in the configuration. (this is fixed in the next release)
So can you please check the e-mail configuration of your instance?
Thank you!

**Comment by nilsnolde** [comment]
> In fact we received like 10 mailserver errors from these e-mails from your instance and got blocked by a few mail providers

yeah I already got notified by someone else. Sorry for that, I only inherited the service as admin recently. I wasn’t aware that your server would be the default (good that that’s changed for the next release!). 

I think the previous years they simply set up an entire new instance so that never happened before. Which I did now too, as resetting is not very straight forwards and I was short on time.

Would you accept a PR to reset an existing instance? (Preferably choosing what to include in the reset and eg exclude certain users) Tbh, I didn’t look at it yet, but should be fairly easy to implement right? If so, I’d ask for some budget to take care of it. For annually recurring events with a decent overlap of both content and users that could be very useful IMO.

**Comment by msquare** [comment]
Interesting idea.

Can you please explain what "asking for some budget" means?
Since we do this project in our personal free time as a hobby there is no money involved.

**Comment by nilsnolde** [comment]
Not your budget😄Just ignore that. 

I’ll get back to it once I confirm FOSSGIS would be interested.

**Comment by msquare** [comment]
:D Okay!

**Comment by MyIgel** [comment]
It could be a feature, but on the other side its "just" one SQL query so imho that could just be added to the [admin documentation](https://github.com/engelsystem/engelsystem-doc/tree/main/content/admin) as "useful queries" or something

**Comment by xuwhite** [comment]
@nilsnolde any update for this issue?
